docs/field-configurations.md

```diff
@@ -9,13 +9,53 @@ They must be added to a file named `configs.yml` in the assets template folder o
 The config file is a yaml file consisting of an array of config entry.
 
 For each config entry the following fields are available:
-- `name` *mandatory*: dotted path field, as in `fields.yml`
+- `name` *mandatory*: dotted path field
 - `fuzziness` *optional (`long` and `double` type only)*: when generating data you could want generated values to change in a known interval. Fuzziness allow to specify the maximum delta a generated value can have from the previous value (for the same field), as a delta percentage; value must be between 0.0 and 1.0, where 0 is 0% and 1 is 100%. When not specified there is no constraint on the generated values, boundaries will be defined by the underlying field type
 - `range` *optional (`long` and `double` type only)*: value will be generated between `min` and `max`
 - `cardinality` *optional*: distribution of different values for the field, expressed as a ratio between a `numerator` and a `denominator`
 - `object_keys` *optional (`object` type only)*: list of field names to generate in a object field type; if not specified a random number of field names will be generated in the object filed type
 - `value` *optional*: hardcoded value to set for the field (any `cardinality` will be ignored)
-- `enum` *optional (`keyword` type only)*: list of strings to randomly chose from a value to set for the field
+- `enum` *optional (`keyword` type only)*: list of strings to randomly chose from a value to set for the field (any `cardinality` will be ignored)
```

Comment:
@aspacca I'm not sure this addition is correct.

Reply:
@endorama indeed not
we generate the values exclusively from the `enum` list, and we apply `cardinality`

what can happens is that the length of the `enum` list is shorter than the `cardinality` values
for example:
```
enum: ["1", "2", "3"]
cardinality:
   numerator: 1
   denominator: 6
```

according to `cardinality` we'll generate a value 6 times, each value will be 1 out of 3 available values in the `enum` list

so the generated values will be repeated and not 6 different ones.
but still once we generate the 6 values (with duplicates) we will emit the 6 values always in the same sequence.

The latest is a details about `cardinality` that's probably worth to clarify (I still have to read your new content, so maybe it's already covered): `cardinality` not only tries to produce a `numerator/denominator` amount of different values[1] but it does in a sequence that will be repeated in a fixed way while emitting the events
so if `numerator/denominator` is 3 and the cardinal sequence generated is "B is a random value", "random value 0", "a different random value", and we emit 7 events, the resulting events will be
1. event: "B is a random value"
2. event: "random value 0"
3. event: "a different random value"
4. event: "B is a random value"
5. event: "random value 0"
6. event: "a different random value"
7. event: "B is a random value"

[1] as much of its possibility, that's not always possible like the case above or if we don't have enough "entropy" to generate all the number of values requested different from each other - imagine asking to generate 20 different integer in the range between 1 and 10, or generating more different keyword type values that the number of different values that we can ask github.com/Pallinder/go-randomdata to generate